File: ios/Classes/Enums/Method Channel/Card Merchant/Mpgs/MethodChannelMpgsGatewayRegion.swift
//
//  MethodChannelMpgsGatewayRegion.swift
//  Runner
//
//  Created by Ray Chow on 29/4/2022.
//

import Foundation

enum MethodChannelMpgsGatewayRegion: String {
    case asiaPacific = "ap",
    europe = "eu",
    northAmerica = "na",
    india = "in",
    mtf = "test",
    cn = "cn"
    
    func convertToMpgsGatewayRegion() -> GatewayRegion {
        switch (self) {
        case .asiaPacific:
            return GatewayRegion.asiaPacific
        case .europe:
            return GatewayRegion.europe
        case .northAmerica:
            return GatewayRegion.northAmerica
        case .india:
            return GatewayRegion.india
        case .mtf:
            return GatewayRegion.mtf
        case .cn:
            return GatewayRegion.china
        }
    }
}

File: android/src/main/kotlin/com/keychainpay/keychainpay_mpgs/enums/method_channel/card_merchant/mpgs/MethodChannelMpgsGatewayRegion.kt
package com.keychainpay.keychainpay_mpgs.enums.method_channel.card_merchant.mpgs

import com.mastercard.gateway.android.sdk.Gateway

enum class MethodChannelMpgsGatewayRegion(val key: String) {
    ASIA_PACIFIC("ap"),
    EUROPE("eu"),
    NORTH_AMERICA("na"),
    INDIA("in"),
    MTF("test"),
    CN("cn")
}

fun MethodChannelMpgsGatewayRegion.convertToMpgsGateInfo(): Gateway.Region {
    return when (this) {
        MethodChannelMpgsGatewayRegion.ASIA_PACIFIC -> Gateway.Region.ASIA_PACIFIC
        MethodChannelMpgsGatewayRegion.EUROPE -> Gateway.Region.EUROPE
        MethodChannelMpgsGatewayRegion.NORTH_AMERICA -> Gateway.Region.NORTH_AMERICA
        MethodChannelMpgsGatewayRegion.INDIA -> Gateway.Region.INDIA
        MethodChannelMpgsGatewayRegion.MTF -> Gateway.Region.MTF
        MethodChannelMpgsGatewayRegion.CN -> Gateway.Region.CHINA
    }
}